kubernetes/cache/cache.go

```diff
@@ -59,15 +58,44 @@ type (
 		proxyStatus *kubernetes.ProxyStatus
 	}
 
+	// cacheLister combines a bunch of lister types into one.
+	// This can probably be simplified or turned into an interface
+	// with go generics.
+	cacheLister struct {
+		// Kube listers
+		configMapLister   core_v1_listers.ConfigMapLister
+		daemonSetLister   apps_v1_listers.DaemonSetLister
+		deploymentLister  apps_v1_listers.DeploymentLister
+		endpointLister    core_v1_listers.EndpointsLister
+		podLister         core_v1_listers.PodLister
+		replicaSetLister  apps_v1_listers.ReplicaSetLister
+		serviceLister     core_v1_listers.ServiceLister
+		statefulSetLister apps_v1_listers.StatefulSetLister
```

Comment:
This doesn't seem to cover all of the workload types we support, is that intended?

Reply:
It's all the types that are currently cached today. The "optional" types such as `DeploymentConfig` and `CronJob` are not cached. I can add those as part of this change but I was planning on adding them in a followup where we remove the `cache_enabled` option and make caching permanently enabled.

Reply:
No problem adding as a follow-up, was just trying to understand.